CONTRIBUTING.md

```diff
@@ -111,6 +113,39 @@ This is the script users can add to their `$PATH` to invoke the CLI more easily
 
 ### Testing
 
+#### Two types of tests
+There are two types of CLI tests:
+- Unit tests call commands directly in Java. They run against source code; no CLI installation is required.
+Example unit test code:
+```
+    // `terra auth status --format=json`
+    TestCommand.Result cmd = TestCommand.runCommand("auth", "status", "--format=json");
+```
+- Integration tests call commands from a bash script run in a separate process. They run against a CLI installation,
+either one built directly from source code via `./gradlew install` or one built from the latest GitHub release.
+Example integration test code:
+```
+    // run a script that includes a Nextflow workflow
+    int exitCode = new TestBashScript().runScript("NextflowRnaseq.sh");
+```
+
+While it's possible to mix both types of testing in the same JUnit method, that should be avoided because then the
+test is running commands against two different versions of the code (the source code directly in the same process
+as the test, and the installed code in a separate process as the test). This could be confusing to track down errors.
+
+Both types of tests:
+- Use the same code to authenticate a test user without requiring browser interaction.
+- Override the context directory to `build/test-context/`, so that tests don't overwrite the context for an existing
```

Comment:
Not sure. I stuck it in there so it would get cleaned up by the `./gradlew clean` task, and that's where other Gradle output lives. Open to any other suggestions though!